src/clients/api/mutations/useExitMarket.ts

```diff
@@ -0,0 +1,29 @@
+import { useMutation, MutationObserverOptions } from 'react-query';
+
+import { exitMarket, IExitMarketInput, ExitMarketOutput } from 'clients/api';
+import FunctionKey from 'constants/functionKey';
+import { useComptroller } from 'hooks/useContract';
+
+const useExitMarket = (
+  // @TODO: use custom error type (see https://app.clickup.com/t/2rvwhnt)
+  options?: MutationObserverOptions<
+    ExitMarketOutput,
+    Error,
+    Omit<IExitMarketInput, 'comptrollerContract'>
+  >,
+) => {
+  const comptrollerContract = useComptroller();
+
+  // @TODO: invalidate related queries on success
```

Comment:
I don't think we need this comment because the PR is open and requires adjustments